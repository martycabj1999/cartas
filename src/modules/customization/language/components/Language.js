import React, { useState, Fragment } from 'react'
import { FormControl, Button, Grid, Select, InputLabel, MenuItem } from '@material-ui/core'
import { makeStyles } from '@material-ui/core/styles';
import { useDispatch, useSelector } from 'react-redux';
import SaveIcon from '@material-ui/icons/Save';

//Action de redux 
import { setLanguageAction } from '../store/LanguageAction';
import { messagesSpanish, messagesEnglish, messagesPortuguese } from '../messages';

const useStyles = makeStyles(theme => ({

    title: {
        fontSize: 24,
    },
    formControl: {
        minWidth: 200,
        //margin: theme.spacing(5),

    },
    button: {
        [theme.breakpoints.down('xs')]:{
            marginTop: theme.spacing(2),
          }
    }

}));

const Language = () => {

    const classes = useStyles();
    const currentLanguage = useSelector(state => state.language.language);
    const messages = useSelector(state => state.language.messages);
    const [language, setLanguage] = useState(currentLanguage);

    //utilizar useDispatch y te crea una funcion
    const dispatch = useDispatch();

    const changeLanguage = (e) => {
        const selectedLanguage = e.target.value;
        setLanguage(selectedLanguage);
    }

    const onSaveLanguage = () => {
        switch (language) {
            case "es":
                let languageES = {
                    language: language,
                    messages: messagesSpanish
                }
                dispatch(setLanguageAction(language, messagesSpanish));
                localStorage.setItem('language', JSON.stringify(languageES))
                break;
            case "en":
                let languageEN = {
                    language: language,
                    messages: messagesEnglish
                }
                dispatch(setLanguageAction(language, messagesEnglish));
                localStorage.setItem('language', JSON.stringify(languageEN))
                break;
            case "pt":
                let languagePT = {
                    language: language,
                    messages: messagesPortuguese
                }
                dispatch(setLanguageAction(language, messagesPortuguese));
                localStorage.setItem('language', JSON.stringify(languagePT))
                break;
            default:
                break;
        }
    }

    return (
        <Fragment>
            <Grid container
                direction="column">
                <Grid
                    container
                    direction="column"
                    justify="center"
                    alignItems="center"
                >
                    <Grid item>
                        <FormControl className={classes.formControl}>
                            <InputLabel id="demo-simple-select-required-label">{messages.language_change}</InputLabel>
                            <Select value={language} onChange={changeLanguage} >
                                <MenuItem value='es'>{messages.language_change_spanish}</MenuItem>
                                <MenuItem value='en'>{messages.language_change_english}</MenuItem>
                                <MenuItem value='pt'>{messages.language_change_portuguese}</MenuItem>
                            </Select>
                        </FormControl>
                    </Grid>
                </Grid>
                <Grid
                    container
                    direction="column"
                    justify="flex-end"
                    alignItems="flex-end"
                >
                    <Grid item className={classes.button}>
                        <Button
                            startIcon={<SaveIcon />}
                            variant="contained"
                            color="primary"
                            onClick={onSaveLanguage}
                        >
                            {messages.color_picker_save}
                        </Button>
                    </Grid>
                </Grid>
            </Grid>
        </Fragment>
    );
}

export default Language;