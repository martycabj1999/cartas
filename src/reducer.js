import { combineReducers } from 'redux';
import AuthReducer from './modules/auth/store/AuthReducer';
import ColorsReducer from './modules/customization/colorPicker/store/ColorsReducer';
import LanguageReducer from './modules/customization/language/store/LanguageReducer';

export default combineReducers({
    security: AuthReducer,
    customization: ColorsReducer, 
    language: LanguageReducer
})