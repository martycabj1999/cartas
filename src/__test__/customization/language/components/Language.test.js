import React from 'react';
import { shallow } from 'enzyme';
import { createStore } from 'redux';
import reducer  from '../../../../reducer';
import { Provider } from 'react-redux';
import Language from '../../../../modules/customization/language/components/Language';

describe("<Language/>", () => {

  const store = createStore(reducer);

  const language = shallow(
    <Provider store={store}>
      <Language />
    </Provider>
  ).dive();

  test("Render del Language", () => {

    expect(language.length).toEqual(1);
    
  });

  test("SnapShot del Language", () => {

    expect(language).toMatchSnapshot();
    
  });

  /*test("Promise updateLanguage", () => {   

    const language = 'es';

    updateLanguage(language).then( response => {
        expect(response.data.length).toBeGreaterThan(0);
    })
  });

  test("Return updateLanguage", () => {   

    const language = 'es';

    updateLanguage(language).then( response => {
        console.log(response.data);
        expect(response.data).toEqual('ndeah');
    })
    .catch( error => {
        console.error(error);
    })
  });*/

});