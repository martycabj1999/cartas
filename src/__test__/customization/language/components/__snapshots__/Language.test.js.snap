// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Language/> SnapShot del Language 1`] = `ShallowWrapper {}`;
